### ISSUE TITLE
[GR] Impossible to get rid of markerstrokes

### ISSUE THREAD

**Issue by baggepinnen** [JuliaPlots/Plots.jl#2494]
These are all attmepts at getting rid of the marker stroke, and they all fail. The stroke gets thinner, but does not dissappear.
```julia
theme(:juno)
scatter(randn(3))
```
![tt2](https://user-images.githubusercontent.com/3797491/77154707-2d5ffe00-6ad7-11ea-8cce-fbd972c1fe47.png)

```julia
theme(:juno)
scatter(randn(3), msc=:transparent)
```
![tt](https://user-images.githubusercontent.com/3797491/77154671-1de0b500-6ad7-11ea-9e1e-48f053485e59.png)

```julia
scatter(randn(3), msw=0)
```
```julia
scatter(randn(3), msa=0)
```

**Comment by daschw** [comment]
Unfortunately GR does not support setting a different alpha for marker and markerstroke.
However, for me `markerstrokewidth` works.
```julia
scatter(rand(10), markersize = 15, markerstrokewidth = 0)
```
![msw](https://user-images.githubusercontent.com/16589944/77157710-2ec30380-6aa2-11ea-9378-6da4d1c637e8.png)

When saving to pdf ([msw.pdf](https://github.com/JuliaPlots/Plots.jl/files/4359134/msw.pdf)) I see the thin stroke issue you described, but not for png and svg.

Another workaround is to use Plots `Shape` as markers. Here you have separate control of marker and markerstroke:
```julia
circ = Shape(Plots.partialcircle(0, 2π))
scatter(rand(10), markersize = 15, markerstrokecolor = nothing, markershape = circ)
scatter(rand(10), markersize = 15, markerstrokealpha = 0, markershape = circ)
```
![markershape](https://user-images.githubusercontent.com/16589944/77158099-e9eb9c80-6aa2-11ea-91ed-71ec35934990.png)

**Comment by baggepinnen** [comment]
You are using a white background, I noticed that the part of the markerstroke that does not go away is always white, no matter the color of the stroke. Maybe you'll see it if your background is dark.

**Comment by daschw** [comment]
```julia
scatter(rand(10), markersize = 15, markerstrokewidth = 0, bginside = :black)
```
![markerdarkbg](https://user-images.githubusercontent.com/16589944/77171817-4956a600-6abd-11ea-9824-b24c0c62aedf.png)

**Comment by daschw** [comment]
Also there is `msc=:auto` to automatically match the marker color.

**Comment by CarloLucibello** [comment]
what about the opposite behavior, i.e. get rid of the marker fill and keep only the stroke? Still trough Shape?

**Comment by mkborregaard** [comment]
That should work on the most recent version, no?

**Comment by daschw** [comment]
No, that's still only possible with `Shape`. AFAIU, using GR's native `GR.polymarker` we cannot set the transparency for border and fill separately. Or is this possible @jheinen?

**Comment by jheinen** [comment]
@daschw : No, transparency would set both border and fill

**Comment by BioTurboNick** [comment]
I just encountered this. It seems to be a change? I didn't used to have a problem with unremovable white borders in GR.

**Comment by jheinen** [comment]
In plain GR, you can set the `borderwidth` to 0, e.g.
```
using GR
setborderwidth(0)
scatter(1:100, randn(100))
```
@daschw: Could this be implemented in `Plots`?

**Comment by daschw** [comment]
We already use that here https://github.com/JuliaPlots/Plots.jl/blob/master/src/backends/gr.jl#L344

**Comment by anneaux** [comment]
> Also there is `msc=:auto` to automatically match the marker color.

i think this doesn't work if markercolor is a color gradient:

<pre><code>import ColorSchemes.turbo  
scatter(rand(10), zcolor= rand(10), ms = 15, mc = :turbo, msc = :auto)
</code></pre>
![test](https://user-images.githubusercontent.com/50271588/93662728-91ea1e80-fa62-11ea-8bd7-fe296451e880.png)

How can I get ride of the strokes in that case?

**Comment by jheinen** [comment]
You might set `markerstrokewidth` to 0:
```
scatter(rand(10), zcolor= rand(10), ms = 15, mc = :turbo, msc = :auto, msw = 0)
```

**Comment by anneaux** [comment]
I tried that already but neither `mswidth = 0` nor `msalpha = 0` were working. As a workaround I used the value for `zcolor` (mapped into [0,1]) as index for the colorscheme, e.g. like: 
<pre><code> 
colorvalues = rand(10)
scatter(rand(10), zcolor=colorvalues, ms = 15, mc =turbo[colorvalues]) </code></pre>

**Comment by jheinen** [comment]
Strange, that's what I get (with `msw = 0`):
<img width="712" alt="Screen Shot 2020-09-21 at 21 47 22" src="https://user-images.githubusercontent.com/5718244/93813946-2667a300-fc54-11ea-8c74-8eb891b7ffed.png">

**Comment by anneaux** [comment]
also when you save as pdf? (for other export formats it works for me as well, but not for pdf)

**Comment by jheinen** [comment]
You are right. For PDF, it doesn't work in `Plots` (although everything work fine in plain `GR`!?) Will have to dive in the `Plots` code again ...

**Comment by t-bltg** [comment]
Setting `msw=0` works for me, saving as pdf:

```julia
using ColorSchemes
using Plots

colorvalues = rand(10)
scatter(rand(10), zcolor=colorvalues, ms=15, mc=ColorSchemes.turbo[colorvalues], msw=0)
Plots.pdf("foo")
```

[foo.pdf](https://github.com/JuliaPlots/Plots.jl/files/6900778/mks.pdf)

**Comment by luboshanus** [comment]
If I plot a `DataFrame` using `StatsPlots` with `gr()` and then use `group` the one group has no stroke but other groups have a hairline stroke somehow. And the group is not actually the first group by which I do groups, it is just the first color used.

<img width="631" alt="Screen Shot 2022-08-30 at 16 58 16" src="https://user-images.githubusercontent.com/2473206/187471600-0874a4c7-a91f-494c-b91a-a9feb665a625.png">

**Comment by zsteve** [comment]
This is still an issue in 2023. Setting `markerstrokewidth = 0` works for me when running in a Jupyter environment, but when using Plots.jl with GR backend  this doesn't seem to make a difference... 

When plotting >1000 points and using a color gradient via `marker_z`, this heavily affects the useability of the package unfortunately. 

```
julia> using Plots
julia> scatter(rand(10), rand(10); markerstrokewidth = 0, markerstrokealpha = 0)
```

![image](https://user-images.githubusercontent.com/4413359/235804993-d6c6d821-d9f0-4d07-af03-38ee05b71123.png)

**Comment by BioTurboNick** [comment]
You know what? `markerstrokewidth = -1` gets it done.

There are a few places where the GR backend has offsets instead of overrides for default values. This must be another one.

**Comment by zsteve** [comment]
> You know what? `markerstrokewidth = -1` gets it done.
> 
> There are a few places where the GR backend has offsets instead of overrides for default values. This must be another one.

Wow!!! It indeed works.

**Comment by luboshanus** [comment]
It does not work for me in julia 1.8.5 while using notebooks and save figure as PDF.
Plots: `[91a5bcdd] Plots v1.38.10`

In the notebook:
<img width="949" alt="jupyter" src="https://user-images.githubusercontent.com/2473206/235871013-5cf39aec-2241-48fd-bb46-3d7fbd9e27af.png">

saved in png:
<img width="1383" alt="in png" src="https://user-images.githubusercontent.com/2473206/235871082-364a027e-38ac-4f54-8655-c6c2ad12232d.png">

saved in pdf:
<img width="1367" alt="in pdf" src="https://user-images.githubusercontent.com/2473206/235871107-3b7a7400-fc39-4a7c-a512-b40f82c6dada.png">
